src/theory/quantifiers/bv_inverter.cpp

```diff
@@ -144,71 +144,83 @@ Node BvInverter::getPathToPv(
   return Node::null();
 }
 
-Node BvInverter::eliminateSkolemFunctions(
-    TNode n, std::vector<Node>& side_conditions,
-    std::unordered_map<TNode, Node, TNodeHashFunction>& visited) {
-  std::unordered_map<TNode, Node, TNodeHashFunction>::iterator it =
-      visited.find(n);
-  if (it == visited.end()) {
-    Trace("bv-invert-debug")
-        << "eliminateSkolemFunctions from " << n << "..." << std::endl;
-    Node ret = n;
-    bool childChanged = false;
-    std::vector<Node> children;
-    if (n.getMetaKind() == kind::metakind::PARAMETERIZED) {
-      children.push_back(n.getOperator());
-    }
-    for (unsigned i = 0; i < n.getNumChildren(); i++) {
-      Node nc = eliminateSkolemFunctions(n[i], side_conditions, visited);
-      childChanged = childChanged || n[i] != nc;
-      children.push_back(nc);
-    }
-    if (childChanged) {
-      ret = NodeManager::currentNM()->mkNode(n.getKind(), children);
-    }
-    // now, check if it is a skolem function
-    if (ret.getKind() == APPLY_UF) {
-      Node op = ret.getOperator();
-      TypeNode tnp = op.getType();
-      // is this a skolem function?
-      std::map<TypeNode, Node>::iterator its = d_solve_var.find(tnp);
-      if (its != d_solve_var.end() && its->second == op) {
-        Assert(ret.getNumChildren() == 1);
-        Assert(ret[0].getType().isBoolean());
+Node BvInverter::eliminateSkolemFunctions(TNode n,
+                                          std::vector<Node>& side_conditions) {
+  std::unordered_map<TNode, Node, TNodeHashFunction> visited;
+  std::unordered_map<TNode, Node, TNodeHashFunction>::iterator it;
+  std::vector<TNode> visit;
```

Comment:
I'd prefer a stack here, mainly for better readability (since it actually is a stack).

Reply:
Done.